rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -242,26 +242,29 @@ func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endB
 			}
 
 			var highestFinalizedBlockNumber uint64
+			batchWriter := s.db.NewBatch()
 			for index := startBatchIndex; index <= batchIndex; index++ {
 				parentBatchMeta, chunks, err := s.getLocalInfoForBatch(index)
 				if err != nil {
 					return fmt.Errorf("failed to get local node info, batch index: %v, err: %w", index, err)
 				}
 
-				endBlock, finalizedBatchMeta, err := validateBatch(event, parentBatchMeta, chunks, s.bc.Config(), s.stack)
+				endBlock, finalizedBatchMeta, err := validateBatch(index, event, parentBatchMeta, chunks, s.bc.Config(), s.stack)
 				if err != nil {
 					return fmt.Errorf("fatal: validateBatch failed: finalize event: %v, err: %w", event, err)
 				}
 
-				rawdb.WriteFinalizedBatchMeta(s.db, index, finalizedBatchMeta)
+				rawdb.WriteFinalizedBatchMeta(batchWriter, index, finalizedBatchMeta)
+				highestFinalizedBlockNumber = endBlock
 
 				if index%100 == 0 {
 					log.Info("finalized batch progress", "batch index", index, "finalized l2 block height", endBlock)
 				}
-
-				highestFinalizedBlockNumber = endBlock
 			}
 
+			if err := batchWriter.Write(); err != nil {
+				log.Crit("Failed to write finalized batch meta to database", "err", err)
```

Comment:
While this is indeed critical, there's no need to use `log.Crit` here, we could just log error and return.

Reply:
sounds good. changed to log.Error with some debugging info in [18cc8d7](https://github.com/scroll-tech/go-ethereum/pull/904/commits/18cc8d7a74fa97375e20bd8a214c0c19fd97bc61).